### DISCUSSION TITLE
Schema's Uploaded to Apicurio are in they're final state?

### DISCUSSION THREAD

**Discussion by jonnyoboy2110** [Apicurio/apicurio-registry#3861] [Q&A] [answered]
Just a quick questions, I cannot find any direct answer in the documentation but from what I can tell, once you upload an artifact to Apicurio, is that version of the schema in now immutable? As in you cannot edit any part of that schema version? If you want to make an update to that schema, a new version will be created?

**Comment by EricWittmann** [comment, marked as answer]
Yes this is correct.  There is a feature of Registry that allows you to delete a version (this functionality must be enabled with an ENV variable when deploying the server).  So if you made some sort of mistake with the version, you could delete it and then re-add it.

Also some of the metadata is editable for a version.  But the content is not.

Originally we were planning on being much more strict about the immutability of a version (do not even allow deleting).  But while it was attractive conceptually, it was not tenable practically for many users.  :)
